lib/child_process.js

```diff
@@ -347,7 +347,11 @@ function normalizeSpawnArguments(file /*, args, options*/) {
     }
   }
 
-  args.unshift(file);
+  if (typeof options.argv0 === 'string') {
```

Comment:
given that this is read-only now, perhaps simply doing an existence check is enough? e.g. 

```js
args.unshift(options.argv0 ? options.argv0 : file);
```